core-common/src/main/java/org/glassfish/jersey/message/internal/OutboundJaxrsResponse.java

```diff
@@ -71,7 +72,7 @@ public static OutboundJaxrsResponse from(javax.ws.rs.core.Response response) {
         } else {
             final StatusType status = response.getStatusInfo();
             final OutboundMessageContext context = new OutboundMessageContext();
-            context.getHeaders().putAll(response.getMetadata());
+            copyEndToEndHeaders(response.getHeaders(), context.getHeaders());
```

Comment:
Should this be ` copyEndToEndHeaders(response.getMetadata(), context.getHeaders());`?

Reply:
I was following the instructions found in the `getMetadata()` javadoc :
```java
    /**
     * See {@link #getHeaders()}.
     *
     * This method is considered deprecated. Users are encouraged to switch their
     * code to use the {@code getHeaders()} method instead. The method may be annotated
     * as {@link Deprecated &#64;Deprecated} in a future release of JAX-RS API.
     *
     * @return response headers as a multivalued map.
     */
    public abstract MultivaluedMap<String, Object> getMetadata();
```